ui/helpers/utils/accounts.js

```diff
@@ -73,43 +74,79 @@ export function getAvatarNetworkColor(name) {
   }
 }
 
-export function getAccountLabel(
+export function getAccountLabels(
   type,
   account,
+  keyrings,
   ///: BEGIN:ONLY_INCLUDE_IF(keyring-snaps)
   snapName,
-  snapPreinstalled,
   ///: END:ONLY_INCLUDE_IF
 ) {
   if (!account) {
-    return null;
+    return [];
   }
+
+  const labels = [];
+
+  const hdKeyrings = keyrings.filter(
+    (keyring) => keyring.type === KeyringType.hdKeyTree,
+  );
+
   switch (type) {
-    case KeyringType.hdKeyTree:
-      return null;
+    case KeyringType.hdKeyTree: {
+      if (hdKeyrings.length > 1) {
+        const hdKeyringIndex = hdKeyrings.findIndex((kr) =>
+          kr.accounts.includes(account.address),
+        );
+        const hdKeyringLabel = `SRP #${hdKeyringIndex + 1}`;
+        labels.push(hdKeyringLabel);
```

Comment:
Nit: I would make a simple a simple helper and use it here and for `KeyringType.snap` too, like:

```ts
const toSrpLabel = (index) => (
  // Index starts at 1, for SRPs.
  `SRP #${index + 1}`
); 
```

This way this would become:
```suggestion
        labels.push(toSrpLabel(hdKeyringIndex));
```